tests/test_frontdoor.py

```diff
@@ -76,6 +76,29 @@ def test_flow_errors():
     assert page.has_one("h2:contains('There is a problem')")
     assert page.has_text("Select where the property is located")
 
+def test_british_gas_unavailable():
+    client = utils.get_client()
+    page = client.get("/start")
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.status_code == 200
+    session_id = page.path.split("/")[1]
+    assert uuid.UUID(session_id)
+
+    form = page.get_form()
+    form["country"] = "England"
+    page = form.submit().follow()
+
+    form = page.get_form()
+    form["supplier"] = "British Gas"
+    page = form.submit().follow()
+
+    assert page.has_text("British Gas are not taking referrals right now but they will be shortly")
+
+    page = page.click(contains="Back")
+    assert page.has_one("h1:contains('Select your home energy supplier from the list below')")
```

Comment:
The NI test above checks the _page_name from the session, wondering if I should be doing that here too? I tried and it failed, couldn't work out exactly how the logic works

Reply:
I think the test as you've written it does the job fine